src/test/java/com/puppycrawl/tools/checkstyle/filters/SuppressWithNearbyTextFilterTest.java

```diff
@@ -0,0 +1,604 @@
+    }
+
+    /**
+     * Calls the filter with a real input file and returns a list of
+     * {@link SuppressWithNearbyTextFilter} internal type {@code Suppression}.
+     *
+     * @return {@code Suppression} list
+     */
+    private static List<?> getSuppressionsAfterExecution(SuppressWithNearbyTextFilter filter,
+                                                         String filename) {
+        final AuditEvent dummyEvent = buildDummyAuditEvent(filename);
+        filter.accept(dummyEvent);
+        return TestUtil.getInternalState(filter, "suppressions");
+    }
+
+    private static String getCurrentFileAbsolutePathAfterExecution(SuppressWithNearbyTextFilter
+                                                                           filter,
+                                                                   String filename) {
+        final AuditEvent dummyEvent = buildDummyAuditEvent(filename);
+        filter.accept(dummyEvent);
+        return TestUtil.getInternalState(filter, "currentFileAbsolutePath");
+    }
+
+    private static AuditEvent buildDummyAuditEvent(String filename) {
+        final Violation violation = new Violation(1, null, null,
+                null, null, Object.class, null);
+        return new AuditEvent("", filename, violation);
+    }
+
+    private static String getLineLengthCheckMessage(int expectedLength, int actualLength) {
+        return getCheckMessage(LineLengthCheck.class, MSG_KEY, expectedLength, actualLength);
+    }
+
+    private static String getRegexpSinglelineCheckMessage() {
+        final String msgRegexExceeded = "regexp.exceeded";
+        return getCheckMessage(RegexpSinglelineCheck.class,
+                msgRegexExceeded, REGEXP_SINGLELINE_CHECK_FORMAT);
+    }
+
+    /**
+     * Performs verification of the file with the given file path using specified configuration
+     * and the array of expected messages. Also performs verification of the config with filters
+     * specified in the input file.
+     * This method is an exact copy of
+     * {@link AbstractModuleTestSupport#verifyWithInlineConfigParser} and the purpose of it is
+     * to be able to call {@link #verifyViolationsSortedActualViolations}. This is a temporary fix
+     * until sorting is implemented in {@link AbstractModuleTestSupport#verifyViolations}.
+     *
+     * @param filePath file path to verify.
+     * @param expectedUnfiltered an array of expected unfiltered config.
+     * @param expectedFiltered an array of expected config with filters.
+     * @throws Exception if exception occurs during verification process.
+     * @noinspection JavadocReference
+     * @noinspectionreason This method is temporary.
+     */
+    @SuppressForbidden
+    private void verifyFilterWithInlineConfigParserSortedActualViolations(String filePath,
+                                                            String[] expectedUnfiltered,
+                                                            String... expectedFiltered)
```

Comment:
move it to abstract class we will start migration to this method soon, as it is more correct way of doing it.

Reply:
I thought we opened https://github.com/checkstyle/checkstyle/issues/12847 for this reason.

Reply:
Yes, but better to place this method in abstract class for ready to be reused.

Reply:
The whole problem comes from `verifyViolations`, I really see no point in adding this method to the abstract class when the problem is tied only to this module. When resolving #12847, we can extend `verifyViolations` with sorting and it should not fail any test in any module.

Reply:
We unlikely be able to meet migrate quickly, it is huge refactoring. We will do this in slow migration as we do for all other migrations: we create better functionality, we violation to use old approach and suppress all violations, we create  epic issue to migrate module by module, epic issue is used to train new contributors. Eventually it is done.

Reply:
Waiting for response to https://github.com/checkstyle/checkstyle/pull/12151#issuecomment-1483840903.

Reply:
Done.